### ISSUE TITLE
Disk graph

### ISSUE THREAD

**Issue by vimalaguti** [ClementTsang/bottom#205]
## Describe the feature request

Hi, I'm usually interested in the IO disk graph but this information is only available in programs like nmon and the windows 10 task manger. 
I would really appreciate a plugin showing MiB read/written and active time, maybe an history graph too - for each drive.

**Comment by ClementTsang** [comment]
@vimalaguti yep, graphing stuff like temps and disk have been something I want to get to --- definitely going to try to add these at some point.

**Comment by magnus-ISU** [comment]
Just want to bump this and say that this would be a very nice feature. The simplest change that could be made to improve this would be to add a total write/read section to the disks widget, as already exists in the process one. I would like to determine how much of FF's seeming 1GB / hour being written is to my actual disk vs. .cache (a ramdisk for me).

If I could read the code I would try to implement it myself since it really shouldn't be that hard, since the real-time data already is available. But I am really lost in this project, sorry

**Comment by ClementTsang** [comment]
Yeah, this is something I still want to do!    Just been really busy with school/work for a while now (especially the last few weeks), but I have a few weeks off so I can hopefully make significant progress in the coming weeks.  Some of the stuff I'm working on now should hopefully make adding future features like this much easier.

> But I am really lost in this project, sorry

Haha, yeah, hopefully what I'm working on right now improves that slightly...

**Comment by magnus-ISU** [comment]
I saw the notes in the refactor issue about improving code reuse and storing widget state locally in each widget. I also have a tough time looking at big rust projects in general, I made a few edits to alacritty but that was a lot of trial and error. I might take another look at it though if I have time, I'm starting school here this week as well.

**Comment by bataliero** [comment]
Vote up for I/O graphs ! It'd be awesome to have an I/O graph for disk  read and write (similar to the existing Network graph).

**Comment by ChillarAnand** [comment]
Any plans to work on this?

**Comment by ClementTsang** [comment]
Yes, planning on working on it when I have some time.

**Comment by gsmitheidw** [comment]
I came here to see if this was on the roadmap. I manage a number of remote windows systems over ssh and I was looking to find something nicer than querying with powershell. I cobbled this together using AI and Go and it sorta works, it's not great code, but better than nothing: 

https://github.com/gsmitheidw/iotop-w

A panel of something like this in btm would be very cool.

**Comment by ClementTsang** [comment]
I might take a look at this and temp graph this weekend. Been busy with a bunch of stuff but I should have some time this weekend...

**Comment by gsmitheidw** [comment]
I think I've a design flaw in mine - Windows doesn't actually have a sense of IO Wait in the same way as Linux. I was incorrect to call it that. Disk queue depth is probably the closest to it in terms of a 'disk pressure' where processes are starved waiting for disk access. In powershell terms something like:

`Get-Counter '\PhysicalDisk(_Total)\Avg. Disk Queue Length'`

**Comment by igxactly** [comment]
I want this too! I'm considering contributing it, if i get some room(time) for implementation.
